### ISSUE TITLE
Changing schedule makes previous backups unavailable

### ISSUE THREAD

**Issue by jdheyburn** [vmware-tanzu/velero#3963]
**What steps did you take and what happened:**
1. Deployed a statefulset with EBS PV to K8s (I am running k8s on AWS)
2. Created a schedule
  - When a schedule is created it also creates a backup (`velero-mybackup-20210716100044`)
3. Modified a property of the schedule, such as either cron or TTL
4. Previous backups are not available at `velero get backup`, but yet still exist in S3 and as snapshots
5. Velero logs out that there are "backups in the backup location that do not exist in the cluster and need to be synced", attempts to sync each one, reports it as being successful - but yet they do not exist

**What did you expect to happen:**

- Expected that previous backups would persist at `velero get backup`
  - Because they do not persist, I cannot restore from them
- Even though they exist in S3, from [this doc](https://velero.io/docs/v1.6/how-velero-works/#object-storage-sync) I would have expected the `BackupStorageLocation` sync to pick it up again
- You can see in the velero logs below that a synchronise from S3 to CRD is trying to take place but it fails silently

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

- `kubectl logs deployment/velero -n velero`: https://gist.github.com/jdheyburn/2a064ef36ef610e0c57d88529d4e29a2#file-velero_logs-txt
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`

```
$ kubectl get backup/velero-mybackup-20210716100044 -n velero
Error from server (NotFound): backups.velero.io "velero-mybackup-20210716100044" not found
$ velero get backup velero-mybackup-20210716100044
An error occurred: backups.velero.io "velero-mybackup-20210716100044" not found
```

- `velero backup logs <backupname>`: 

```
$ velero backup logs velero-mybackup-20210716100044
Backup "velero-mybackup-20210716100044" does not exist.
```

**Anything else you would like to add:**

schedule.yaml - https://gist.github.com/jdheyburn/2a064ef36ef610e0c57d88529d4e29a2#file-schedule-yaml

I am using the helm-chart (2.23.3) to manage resources, it's values: https://gist.github.com/jdheyburn/2a064ef36ef610e0c57d88529d4e29a2#file-helm-chart-values-yaml

**Environment:**

- Velero version (use `velero version`): 

```
Client:
        Version: v1.6.1
        Git commit: -
Server:
        Version: v1.6.1
```

- Velero features (use `velero client config get features`):  `features: <NOT SET>`

- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.9", GitCommit:"94f372e501c973a7fa9eb40ec9ebd2fe7ca69848", GitTreeState:"clean", BuildDate:"2020-09-16T13:56:40Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.9", GitCommit:"9dd794e454ac32d97cde41ae10be801ae98f75df", GitTreeState:"clean", BuildDate:"2021-03-18T01:00:06Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
```

- Kubernetes installer & version: kops
- Cloud provider or hardware configuration: AWS
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by jdheyburn** [comment]
The same issue occurred when changing the `includeNamespace` setting - in my case it reproduced when adding a new namespace to the list.

**Comment by jdheyburn** [comment]
@reasonerjt Let me know if you need help with this - it's an issue that we see whenever we perform deployments of Velero, and can cause us to lose our backup history.

**Comment by jdheyburn** [comment]
I think I found another means of reproducing:

1. Deploy Velero and create schedule
2. Change a property about the schedule and apply, ensuring at least 1 iteration of the backup schedule has occurred
3. Perform a redeploy of Velero
4. The logs below should continually appear:

```text
time="2021-09-21T09:06:42Z" level=info msg="Found 3 backups in the backup location that do not exist in the cluster and need to be synced" backupLocation=default controller=backup-sync logSource="/Users/joseph.heyburn/code/personal/velero/pkg/controller/backup_sync_controller.go:204"
```

I've enabled a debugging session and paying attention to this [block of code](https://github.com/vmware-tanzu/velero/blob/main/pkg/controller/backup_sync_controller.go#L232-L243) in `backup_sync_controller.go`. It seems creating the Backup resource via the API is silently failing.
